packages/pancake-uikit/src/hooks/useTooltip/useTooltip.tsx

```diff
@@ -58,9 +66,12 @@ const useTooltip = (
       setVisible(true);
       if (trigger === "hover") {
         if (e.target === targetElement) {
-          clearTimeout(tooltipHoverTimeoutRef.current);
-        } else if (e.target === tooltipElement) {
-          tooltipHoverRef.current = true;
+          // If we were about to close the tooltip and got back to it
+          // then prevent closing it.
+          clearTimeout(hideTimeout.current);
+        }
+        if (e.target === tooltipElement) {
```

Comment:
else if here is better I think. What do you think?

Reply:
It doesn't really matter actually since its impossible for both conditions to be true at the same time (since target can't be 2 things at the same time). So mostly matter of taste, I personally think its easier to read without `else if` 🙂

Reply:
Yes, true. I said it because of redundant check in runtime but if it is better readability I agree to stay it like this instead of little performance gain

Reply:
Hm.... that's actually a good point, thanks! 👍 
But yeah, maybe better to not over-optimize stuff for now for the sake of code readability, and there are probably other pieces of code that need optimization much more than this 🙂 

I'm thinking of getting some really cheap phone and laptop someday to see how it all works on low-power devices, easy to forget that most users don't have latest and greatest hardware.

Reply:
I am also a fan of avoiding `else if` at all cost. Easier to read, less bugs, easier to test.